docs/runtime/go-metrics.md

```diff
@@ -0,0 +1,221 @@
+
+**[1]:** Computed from `/gc/heap/allocs:objects`.
+<!-- endsemconv -->
+
+<!-- semconv metric.go.memory.allocations(full) -->
+<!-- endsemconv -->
+
+## Go Garbage Collection
+
+**Description:** Go metrics captured under the namespace `go.memory.gc.*`
+
+### Metric: `go.memory.gc.goal`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.go.memory.gc.goal(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.memory.gc.goal` | UpDownCounter | `By` | Heap size target for the end of the GC cycle. [1] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+**[1]:** Computed from `/gc/heap/goal:bytes`.
+<!-- endsemconv -->
+
+<!-- semconv metric.go.memory.gc.goal(full) -->
+<!-- endsemconv -->
+
+## Go Goroutines
+
+**Description:** Go metrics captured under the namespace `go.goroutine.*`
+
+### Metric: `go.goroutine.count`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.go.goroutine.count(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.goroutine.count` | UpDownCounter | `{goroutine}` | Count of live goroutines. [1] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+**[1]:** Computed from `/sched/goroutines:goroutines`.
+<!-- endsemconv -->
+
+<!-- semconv metric.go.goroutine.count(full) -->
+<!-- endsemconv -->
+
+## Go Threads
+
+**Description:** Go metrics captured under the namespace `go.thread.*`
+
+### Metric: `go.thread.limit`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.go.thread.limit(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.thread.limit` | UpDownCounter | `{thread}` | The number of OS threads that can execute user-level Go code simultaneously. [1] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+**[1]:** Computed from `/sched/gomaxprocs:threads`.
+<!-- endsemconv -->
+
+<!-- semconv metric.go.thread.limit(full) -->
+<!-- endsemconv -->
+
+## Go Scheduler
+
+**Description:** Go metrics captured under the namespace `go.schedule.*`
+
+### Metric: `go.schedule.duration`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.go.schedule.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.schedule.duration` | Histogram | `s` | The time goroutines have spent in the scheduler in a runnable state before actually running. [1] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

Comment:
for any histograms, you'll want to specify the default buckets, check out https://github.com/open-telemetry/semantic-conventions/issues/316#issuecomment-1783528869

Reply:
Bucket boundaries are provided by the runtime, so we can't actually change them via views.  They appear to be:

* 6.399999999999999e-08
* 6.399999999999999e-07
* 7.167999999999999e-06
* 8.191999999999999e-05
* 0.0009175039999999999
* 0.010485759999999998
* 0.11744051199999998

Reply:
I'm asking if those are something that we can rely on: https://github.com/golang/go/issues/67120#issuecomment-2100709396

Reply:
Buckets are not stable, and will change across go versions.  Should we still document them here?

Reply:
Bucket changes are allowed in stable semconv (https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md).

Btw, how are you planning to emit pre-aggregated buckets?

I think the semconv story is a bit murky when it's not treated as a specification, but rather a bridging of an underlying metric system, e.g. check out https://github.com/open-telemetry/semantic-conventions/issues/269#issuecomment-1686756064

Reply:
We were planning to expose histogram metrics using the MetricProducer interface.  This is something in-between a bridge to an underlying metric system and instrumentation.  Go runtime metrics are fairly verbose, and lack attributes, so there is at least some curation and transformation that is needed to make the metrics useful to OTel users.  OTOH, we are definitely re-exposing metric data here.

Reply:
Makes sense, and very similar to the JVM metrics story, I guess we just haven't dealt with preaggregated buckets before. Do you have the option to get individual GC durations that could then be pushed into our own shaped histogram? (Also to allow metric view buckets to work)

Reply:
We don't have that option today.

Reply:
Maybe then just doc the situation in semconv, explains that this histogram has preaggregated buckets from the underlying data source. 

Also not sure but may be worth mentioning that metric views won't work, since that may be a common confusion.

Reply:
Documented here: https://github.com/open-telemetry/semantic-conventions/pull/981/commits/7b0717230641508d33896c2e18c553fbedff091d.  We can document that views won't work in the documentation for the package.